lib/rate_limiting.rb

```diff
@@ -108,7 +108,9 @@ def allowed?(request)
 
   def find_matching_rule(request)
     @rules.each do |rule|
-      return rule if request.path =~ rule.match
+      if rule.match?(request)
```

Comment:
This is a nit pick, but I think this would be easier to read if put back into a one line if statement